src/tiled/objecttypesmodel.cpp

```diff
@@ -91,8 +97,17 @@ bool ObjectTypesModel::setData(const QModelIndex &index,
                                int role)
 {
     if (role == Qt::EditRole && index.column() == 0) {
-        mObjectTypes[index.row()].name = value.toString().trimmed();
+        const int oldRow = index.row();
+        mObjectTypes[oldRow].name = value.toString().trimmed();
         emit dataChanged(index, index);
+
+        auto nextObjectType = std::lower_bound(mObjectTypes.begin(), mObjectTypes.end(), mObjectTypes[oldRow], objectTypeLessThan);
```

Comment:
Hmm, I think there may be problems here because due to the above modification, `mObjectTypes` may no longer be sorted, which could throw off`std::lower_bound`. Maybe just make a local copy first.